src/solvers/mp/RotationMP.h

```diff
@@ -0,0 +1,252 @@
+        completeWeekendCost_(rotation.completeWeekendCost_), preferenceCost_(rotation.preferenceCost_), initRestCost_(rotation.initRestCost_),
+        dualCost_(rotation.dualCost_), timeDuration_(rotation.timeDuration_)
+    {
+      if(rotation.nurseId_ != nurseId_){
+        cost_ = DBL_MAX;
+        dualCost_ = DBL_MAX;
+      }
+    }
+
+    ~Rotation(){}
+
+    int getShift(int day) const override {
+      return shifts_.at(day);
+    }
+
+    //count rotations
+    //
+    static unsigned int s_count;
+
+    // Shifts to be performed
+    //
+    std::map<int,int> shifts_;
+
+    //Id of the rotation
+    //
+    long id_;
+
+    // Cost
+    //
+    double cost_;
+    double consShiftsCost_ , consDaysWorkedCost_, completeWeekendCost_, preferenceCost_, initRestCost_ ;
+
+    // Dual cost as found in the subproblem
+    //
+    double dualCost_;
+
+    // Level of the branch and bound tree where the rotation has been generated
+    //
+    int treeLevel_=0;
+
+    // Time duration (in a certain unit: day, hours, half-hours, ...)
+
+    int timeDuration_;
+
+    //compact the rotation in a vector
+    std::vector<double> getCompactPattern() const override {
+      std::vector<double> pattern = Pattern::getCompactPattern();
+      for(std::pair<int,int> p: shifts_) pattern.push_back(p.second);
+      pattern.push_back(timeDuration_);
+      return pattern;
+    }
+
+    //Compute the cost of a rotation
+    //
+    void computeCost(Scenario* pScenario, const std::vector<LiveNurse*>& liveNurses, int horizon);
+
+    //Compute the dual cost of a rotation
+    //
+    void checkDualCost(DualCosts& costs);
+
+    // calcule le nombre d'heures d'une rotation
+    void computeTimeDuration(Scenario* pScenario) {
+      timeDuration_ = 0;
+      for (std::pair<int, int> p: shifts_) {
+        timeDuration_ += pScenario->timeDurationToWork_[p.second];
+      }
+    }
+
+    std::string toString(int nbDays = -1, std::vector<int> shiftIDToShiftTypeID={}) const override;
+
+    //Compare rotations on index
+    //
+    static bool compareId(const Rotation& rot1, const Rotation& rot2);
+
+    //Compare rotations on cost
+    //
+    static bool compareCost(const Rotation& rot1, const Rotation& rot2);
+
+    //Compare rotations on dual cost
+    //
+    static bool compareDualCost(const Rotation& rot1, const Rotation& rot2);
+};
+
+//-----------------------------------------------------------------------------
+//
+//  C l a s s   M a s t e r P r o b l e m
+//
+// Build and solve the master problem of the column generation scheme
+//
+//-----------------------------------------------------------------------------
+class RotationMP: public MasterProblem {
+  public:
+    RotationMP(Scenario* pScenario, Demand* pDemand, Preferences* pPreferences, std::vector<State> *pInitState,
+               MySolverType solver);
+    virtual ~RotationMP();
+
+    PPattern getPattern(const std::vector<double>& pattern) const override;
```

Comment:
Nice, did not know the override identifier. Very safe programming :)

Reply:
It's a clangy tips from CLion